docs/setup/settings.asciidoc

```diff
@@ -33,14 +33,20 @@ Specify the position of the subdomain the URL with the token `{s}`.
 `elasticsearch.username:` and `elasticsearch.password:`:: If your Elasticsearch is protected with basic authentication,
 these settings provide the username and password that the Kibana server uses to perform maintenance on the Kibana index at
 startup. Your Kibana users still need to authenticate with Elasticsearch, which is proxied through the Kibana server.
-`server.ssl.cert:` and `server.ssl.key:`:: Paths to the PEM-format SSL certificate and SSL key files, respectively. These
-files enable SSL for outgoing requests from the Kibana server to the browser.
+`server.ssl.enabled`:: *Default: "false"* Enables SSL for outgoing requests from the Kibana server to the browser. When set to `true`, `server.ssl.certificate` and `server.ssl.key` are required
+`server.ssl.certificate:` and `server.ssl.key:`:: Paths to the PEM-format SSL certificate and SSL key files, respectively.
+`server.ssl.keyPassphrase`:: The passphrase that will be used to decrypt the private key. This value is optional as the key may not be encrypted.
+`server.ssl.certificateAuthorities`:: List of paths to PEM encoded certificate files that should be trusted.
+`server.ssl.clientAuthentication`:: *Default: none* Controls Kibana's server behavior in regard to requesting a certificate from client connections. Valid values are `required`,
```

Comment:
I've been able to consistently get this log message to occur by setting `server.ssl.clientAuthentication: required` and then opening up Kibana and letting the browser sit on the prompt for me to select a certificate for a long-time. However, I'm sure that other situations could also cause this to occur as well.

This error is being emitted by the [tls socket](https://github.com/nodejs/node/blob/v6.9.4/lib/_tls_wrap.js#L848) to the [https Server](https://github.com/nodejs/node/blob/v6.9.4/lib/_tls_wrap.js#L848) to the [hapi connection](https://github.com/hapijs/hapi/blob/v14.2.0/lib/connection.js#L94) which we then [intercept] (https://github.com/elastic/kibana/blob/master/src/server/logging/log_interceptor.js), and downgrade certain errors.

We've been downgrading errors based on their `errno`, but those only exist on (System Errors)[https://nodejs.org/api/errors.html#errors_system_errors], and these are just generic Errors.

We could potentially filter these out by matching the err.message of `TLS handshake timeout`, but that feels rather brittle. Additionally, there are various other generic client errors the underlying ssl library is throwing with uglier messages like the following:

```
4330128320:error:14094418:SSL routines:ssl3_read_bytes:tlsv1 alert unknown ca:../deps/openssl/openssl/ssl/s3_pkt.c:1487:SSL alert number 48
4330128320:error:140940E5:SSL routines:ssl3_read_bytes:ssl handshake failure:../deps/openssl/openssl/ssl/s3_pkt.c:1210:
```

I'm beginning to wonder whether we should be downgrading all client errors to debug/info. What are your alls thoughts @spalger @tylersmalley @jbudz?